File: updateEnvs.py
import json
import requests
from requests import Response


def conDict(resp: Response) -> dict | list:
    res: dict = json.loads(resp.text)
    return res['data']


def getToken(url: str, client_id: str, client_secret: str) -> str:
    """
    获取青龙面板的token
    :param url: 请求路径
    :param client_id: 密钥id
    :param client_secret: 密钥
    :return: 以字符串形式返回
    """
    resp: Response = requests.get(f'{url}/open/auth/token', params={
        "client_id": client_id,
        "client_secret": client_secret
    })
    data: dict = conDict(resp)
    return f"{data['token_type']} {data['token']}"


def getEnvs(url: str, token: str) -> list:
    """
    获取青龙面板所有的环境变量
    :param url:请求路径
    :param token:青龙面板token
    :return:以字典形式返回
    """
    resp: Response = requests.get(f'{url}/open/envs', headers={
        "Accept": "application/json",
        "Authorization": token
    })
    return conDict(resp)


def updateEnvs(id: int, url: str, token: str, value: str) -> dict:
    """
    更新青龙面板JD_COOKIE环境变量
    :param id:更新环境变量的id
    :param url:请求路径
    :param token:青龙面板token
    :param value:环境变量的value
    :return:
    """
    resp: Response = requests.put(f"{url}/open/envs", data=json.dumps({
        "value": value,
        "name": "JD_COOKIE",
        "remarks": "京东Cookie",
        "id": id
    }), headers={
        "Accept": "application/json",
        "Content-Type": "application/json",
        "Authorization": token
    })
    return json.loads(resp.text)


def main() -> None:
    sourceUrl: str = ''
    sourceClientID: str = ''
    sourceClientSecret: str = ''

    distUrl: str = ''
    distClientID: str = ''
    distClientSecret: str = ''

    global qlId
    sourceToken: str = getToken(
        sourceUrl, sourceClientID, sourceClientSecret)
    cookieDict = getEnvs(sourceUrl, sourceToken)
    cookies: list[str] = [item['value'] for item in cookieDict]
    cookieStr: str = '&'.join(cookies)
    distToken: str = getToken(
        distUrl, distClientID, distClientSecret)
    cookiesInfo: list = getEnvs(distUrl, distToken)

    for item in cookiesInfo:
        if item['name'] == 'JD_COOKIE':
            qlId = item['id']
    res: dict = updateEnvs(qlId, distUrl, distToken, cookieStr)

    if res['code'] == 200:
        print('更新JD_COOKIE环境变量成功!')
    else:
        print('更新JD_COOKIE环境变量失败!')


if __name__ == '__main__':
    main()

File: updateEnvs.ts
import axios from 'axios';

const sourceUrl = '';
const sourceClientID = '';
const sourceClientSecret = '';

const distUrl = '';
const distClientID = '';
const distClientSecret = '';

!(async () => {
    const sourceToken = await getToken(sourceUrl, sourceClientID, sourceClientSecret);
    const cookies: string[] = [];

    const cookieObject = await getEnvs(sourceUrl, sourceToken);
    for (const item of cookieObject as any) {
        cookies.push(item.value);
    }

    const cookieStr = cookies.join('&');

    const distToken = await getToken(distUrl, distClientID, distClientSecret);

    const cookiesInfo = await getEnvs(distUrl, distToken);

    let id: any;

    for (const item of cookiesInfo as any) {
        if (item.name === 'JD_COOKIE') {
            id = item.id;
        }
    }

    const res = await updateEnvs(id as number, distUrl, distToken, cookieStr);

    if ((res as any).code === 200) {
        console.log('更新JD_COOKIE环境变量成功!');
    } else {
        console.log('更新JD_COOKIE环境变量失败!');
    }


})()

// 获取青龙面板登录token
async function getToken(url: string, client_id: string, client_secret: string): Promise<string> {

    const {
        data: res
    } = await axios.get(`${url}/open/auth/token`, {
        params: {
            "client_id": client_id,
            "client_secret": client_secret
        }
    });

    return `${res.data['token_type']} ${res.data['token']}`;
}

async function getEnvs(url: string, token: string): Promise<object> {

    const {
        data: res
    } = await axios.get(`${url}/open/envs`, {
        headers: {
            "Accept": "application/json",
            "Authorization": token
        }
    });

    return res.data;
}

// 更新青龙面板JD_COOKIE环境变量
async function updateEnvs(id: number, url: string, token: string, value: string): Promise<object> {
    const {
        data: res
    } = await axios.put(`${url}/open/envs`, {
        value: value,
        name: "JD_COOKIE",
        remarks: '京东Cookie',
        id: id
    }, {
        headers: {
            "Accept": "application/json",
            "Content-Type": "application/json",
            "Authorization": token
        }
    });
    return res;
}